tests/utils/yaml.go

```diff
@@ -0,0 +1,91 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+)
+
+// CreateObjectsFromYAML creates kubernetes objects defined in a YAML file, in the namespace
+// NOTE: this function is meant to be used in Tests. It only recognizes a subset of
+//  kubernetes and CNPG object types, throwing errors for any other object types
+func CreateObjectsFromYAML(yamlPath, namespace string) ([]client.Object, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(yamlPath))
+	if err != nil {
+		return nil, err
+	}
+
+	sections := bytes.Split(data, []byte("---"))
```

Comment:
If I remember correctly `kubectl apply` is not handling the YAML stream splitting manually but relying on the YAML parser to do that. Perhaps we could explore a bit what's happening here:

https://github.com/kubernetes/kubernetes/blob/aea9f9887d08922378b9e7da0e81a46e733f2995/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L401

to see if we can get some ideas?

Reply:
This patch is already too big! Let's do this later. In the end this works